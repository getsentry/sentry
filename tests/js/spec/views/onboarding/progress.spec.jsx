import React from 'react';
import {shallow} from 'enzyme';

import ProgressNodes from 'app/views/onboarding/progress';

describe('ProgressNodes', function() {
  describe('render()', function() {
    const baseProps = {
      params: {
        projectId: '',
      },
    };

    it('should render step 0 if no projectId', function() {
      let baseContext = {
        context: {
          organization: {id: '1337', slug: 'testOrg', experiments: {}},
          location: {pathname: 'http://onboarding/lol/', query: {}},
        },
      };
      let wrapper = shallow(<ProgressNodes {...baseProps} />, baseContext);

      expect(wrapper.find('.node')).toHaveLength(6);
      expect(wrapper.find('.active')).toHaveLength(2);
      expect(
        wrapper
          .find('.active')
          .first()
          .last()
          .text()
      ).toEqual('Tell us about your project');

      expect(wrapper.find('.done')).toHaveLength(2);

      expect(wrapper).toMatchSnapshot();
    });

    it('should render step 1 if has projectId', function() {
      let baseContext = {
        context: {
          organization: {id: '1337', slug: 'testOrg', experiments: {}},
          location: {
            pathname: 'http://onboarding/lol/projectSlug/configure/platform/',
            query: {},
          },
        },
      };
      let props = {
        ...baseProps,
        params: {
          projectId: 'my-cool-project',
        },
      };

      let wrapper = shallow(<ProgressNodes {...props} />, baseContext);

      expect(wrapper.find('.node')).toHaveLength(6);
      expect(wrapper.find('.active')).toHaveLength(2);
      expect(
        wrapper
          .find('.active')
          .first()
          .last()
          .text()
      ).toEqual('Configure your application and send an event');

      expect(wrapper.find('.done')).toHaveLength(4);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
