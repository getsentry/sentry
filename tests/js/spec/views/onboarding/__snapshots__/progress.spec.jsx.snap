// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressNodes render() should render step 0 if no projectId 1`] = `
<div
  className="onboarding-sidebar"
>
  <div
    className="sentry-flag"
  >
    <span
      className="icon-sentry-logo-full"
      href="/"
    />
  </div>
  <div
    className="progress-nodes"
  >
    <div
      className="node done"
    >
      <span
        className="node done"
      />
      Create an organization in Sentry
    </div>
    <div
      className="node active"
    >
      <span
        className="node active"
      />
      Tell us about your project
    </div>
    <div
      className="node"
    >
      <span
        className="node"
      />
      Configure your application and send an event
    </div>
  </div>
  <div
    className="stuck"
  />
</div>
`;

exports[`ProgressNodes render() should render step 1 if has projectId 1`] = `
<div
  className="onboarding-sidebar"
>
  <div
    className="sentry-flag"
  >
    <span
      className="icon-sentry-logo-full"
      href="/"
    />
  </div>
  <div
    className="progress-nodes"
  >
    <div
      className="node done"
    >
      <span
        className="node done"
      />
      Create an organization in Sentry
    </div>
    <div
      className="node done"
    >
      <span
        className="node done"
      />
      Tell us about your project
    </div>
    <div
      className="node active"
    >
      <span
        className="node active"
      />
      Configure your application and send an event
    </div>
  </div>
  <div
    className="stuck"
  />
</div>
`;
